### PR TITLE
Make addon-options work with story parameters

### PR DESCRIPTION
**Pull request by nm123github**
## What I did
- use makeDectorator to upgrade withOptions
- update documentation

## How to test
I've added usage to some stories in the marko-cli

addon-options-eg

If your answer is yes to any of these, please make sure to include it in your PR.

For maintainers only: Please tag your pull request with at least one of the following:
`["cleanup", "BREAKING CHANGE", "feature request", "bug", "documentation", "maintenance", "dependencies", "other"]`

### CODE REVIEW

**Comment by nm123github** [comment]
Attempting to make addon-options use parameters. Few things,

- I have not deprecated the global setOptions. I think its probably a good idea to let it be there but would like to know what others think.
- Right now when you set options (using params) for a story, they get set globally. Ideally i would probably like it to be reset or something (maybe to the options we set using setOptions?) if we switch to a story where we don't set options. Is it possible to setOptions specifically for a story?

**Review by ndelangen — APPROVED** [review]
SUPER AWESOME!

**Comment by ndelangen** [comment]
setOptions being persistent is expected behavior I think?

I don't see this as a big problem.

File: addons/options/src/preview/index.js [hunk 1]
```diff
@@ -41,3 +41,32 @@ export function setOptions(newOptions) {
 
   channel.emit(EVENT_ID, { options });
 }
+
+export const withOptions = makeDecorator({
+  name: 'withOptions',
+  parameterName: 'options',
+  skipIfNoParametersOrOptions: false,
+  allowDeprecatedUsage: true,
```
Comment (tmeasday) [inline addons/options/src/preview/index.js]: @nm123github -- was there a particular reason why you decided to allow deprecated usage for this decorator?

Comment (nm123github) [reply addons/options/src/preview/index.js]: Just noticed bunch of other add-ons that had set it to true, so presumed this was the pattern to follow and maybe there was a plan to set them to false at a later date.

Comment (tmeasday) [reply addons/options/src/preview/index.js]: Gotcha. That option is to allow `add('story', withOptions(() => <Story />))` style usage (which is the old way addons did it pre-4.0). Given this API never existed for `withOptions` I don't think we need it.

**Pull request merged**